src/common/data/serverActions/getUserData.ts

```diff
@@ -0,0 +1,20 @@
+'use server'
+
+import { createClient } from '@/features/auth/supabase/server'
+import { User } from '@supabase/supabase-js'
+
+interface GetUserDataActionProps {
+  user?: User
+  error?: string
+}
+
+export async function getUserData(): Promise<GetUserDataActionProps> {
+  const supabase = createClient()
+  const { data, error } = await supabase.auth.getUser()
+
+  if (error || !data?.user) {
+    return { error: 'Internal error' }
+  }
+
+  return { user: data.user }
```

Comment:
We should probably somehow reuse the createResponse util that I created for the API responses. How do we set HTTP response codes from server actions?

Reply:
hmm yeah a standardized response could be a nice thing for server actions... though i'm not sure.

a counter argument though to this is that in practice they seem like standard async functions and since they are actually part of the app code you can leverage typescript and IDE intelli-sense and you should know the exact signature of the function.  it's not like http where you need to rely on a standardized protocol.

i think we should let this be for now and discuss later

Reply:
server actions are weird, but very cool.  i guess i was half treating it like an HTTP request and half like a standard async function.  maybe instead of return an error property in the response it makes more sense to just throw an error since we should be try/catching async functions anyway

Reply:
Yeah, under the hood, they are HTTP requests, but Next.js is hiding that implementation detail from us. You are right in that they are fundamentally different with error handling.

https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations#error-handling

We don't have to do it now, but eventually, we would want to DRY up our code and make some reusable handler for all the server actions (which should probably be a different one from the API createResponse version).